app/src/org/commcare/views/widgets/RecordingFragment.java

```diff
@@ -0,0 +1,330 @@
+package org.commcare.views.widgets;
+
+import android.app.Activity;
+import android.content.pm.ActivityInfo;
+import android.content.res.Configuration;
+import android.graphics.Rect;
+import android.media.MediaPlayer;
+import android.media.MediaRecorder;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Environment;
+import android.os.SystemClock;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
+import android.widget.Button;
+import android.widget.Chronometer;
+import android.widget.ImageButton;
+import android.widget.LinearLayout;
+import android.content.DialogInterface;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+import org.commcare.dalvik.R;
+import org.commcare.logging.analytics.GoogleAnalyticsUtils;
+import org.javarosa.core.services.locale.Localization;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Created by Saumya on 6/15/2016.
+ * A popup dialog fragment that handles recording_fragment and saving of audio files without external callout
+ */
+
+public class RecordingFragment extends android.support.v4.app.DialogFragment{
+
+    private String fileName;
+    private final String FILE_EXT = "/Android/data/org.commcare.dalvik/temp/Custom Recording.mp4";
+
+    private LinearLayout layout;
+    private ImageButton toggleRecording;
+    private ImageButton discardRecording;
+    private Button saveRecording;
+    private Button recordAgain;
+    private TextView instruction;
+    private ProgressBar recordingProgress;
+
+    private Chronometer recordingDuration;
+    private long currentTimeMillis;
+
+    private MediaRecorder recorder;
+    private RecordingCompletionListener listener;
+    private MediaPlayer player;
+    
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState){
+        layout = (LinearLayout) inflater.inflate(R.layout.recording_fragment, container);
+        disableScreenRotation();
+        prepareButtons();
+        prepareText();
+        setWindowSize();
+        fileName = Environment.getExternalStorageDirectory().getAbsolutePath() + FILE_EXT;

```

Comment:
If you have two audioprototype widgets, this looks like it would record over the first one?